cpp/src/join/sort_merge_join.cu

```diff
@@ -531,20 +757,111 @@ sort_merge_join::inner_join(table_view const& left,
       preprocessed_left._null_processed_table_view = left;
     }
   }
-  if (is_left_sorted == cudf::sorted::NO) { preprocessed_left.get_sorted_order(stream); }
+  if (is_left_sorted == cudf::sorted::NO) { preprocessed_left.compute_sorted_order(stream); }
 
   return invoke_merge(
     preprocessed_right._null_processed_table_view,
     preprocessed_left._null_processed_table_view,
     [this, stream, mr](auto& obj) {
-      auto [preprocessed_right_indices, preprocessed_left_indices] = obj(stream, mr);
+      auto [preprocessed_right_indices, preprocessed_left_indices] = obj.inner(stream, mr);
       postprocess_indices(*preprocessed_right_indices, *preprocessed_left_indices, stream);
-      stream.synchronize();
       return std::pair{std::move(preprocessed_left_indices), std::move(preprocessed_right_indices)};
     },
     stream);
 }
 
+std::pair<std::unique_ptr<rmm::device_uvector<size_type>>,
+          std::unique_ptr<rmm::device_uvector<size_type>>>
+sort_merge_join::left_join(table_view const& left,
+                           sorted is_left_sorted,
+                           rmm::cuda_stream_view stream,
+                           rmm::device_async_resource_ref mr)
+{
+  cudf::scoped_range range{"sort_merge_join::left_join"};
```

Comment:
`CUDF_FUNC_RANGE` if profiling the entire function.

Reply:
This is intentional: with the macro, it will simply appear as `left_join`, and we can’t tell whether it refers to the free-function `left_join`, `hash_join::left_join`, or `sort_merge_join::left_join`.